helper_test.go

```diff
@@ -11,7 +11,7 @@ func assertErr(t *testing.T, err error) {
 
 func assertEq(t *testing.T, msg string, exp interface{}, act interface{}) {
 	t.Helper()
-	if exp != act {
+	if exp != act && exp != nil && act != nil {
```

Comment:
This condition, if `exp` is nil and `act` is not nil ( or `exp` is not nil and `act` is nil ), it will be false.

Reply:
I look at that now and know what i wanted to do but am puzzled at what i wrote.

I wanted to explicitly check if both values are nil, in which case they are to be considered equal.

Reason for this explicit check is, that comparing a nil literal to a nil of type func does not work when they are hidden behind interface{}. Thus the test will fail even though both values are nil.

Reply:
I think `if exp != act && !(exp == nil && act == nil) ` would solve this.

Reply:
After further testing it seems that hiding the `func()` type behind `interface{}` requires you to get the actual type back before you can compare it to `nil`.
I've now inlined the comparison to `nil` and instead call `assertEq(t, "nilfunc", true, v==nil)` so that the test works as intended.